indico/web/forms/fields/simple.py

```diff
@@ -177,6 +177,12 @@ class IndicoMultipleTagSelectField(SelectMultipleField):
 class IndicoLinkListField(JSONField):
     widget = JinjaWidget('forms/link_list_widget.html', single_kwargs=True, single_line=True)
 
+    def process_formdata(self, valuelist):
+        super().process_formdata(valuelist)
+        self.data = [link for link in self.data if link.get('title') or link.get('url')]
```

Comment:
Why is the filtering needed? Shouldn't we always have link and url keys in the data?

Reply:
Because of the changes to the component (to make values persist on refresh) we can't really filter empty entries client-side anymore without potentially losing info when refreshing. So I moved the filtering to the server-side

Reply:
Shouldn't client-side validation fail and prevent submission in that case? And when you do submit with an empty URL then the normal validation should fail.

Reply:
Oh I see, the "save now" corner button works even if some fields are invalid...

Reply:
There is also the case of when the field is empty - there is always one element on the links list. I could change that but it may complicate the component logic a bit

I also don't see a reason to bother the user if there are empty entries on the list. I don't see what harm it would cause to "silently" delete them

Reply:
When I tested it yesterday it felt all fine, so I think we can leave it as it is